pkg/bootstrap/docker/openshift/helper.go

```diff
@@ -709,6 +723,92 @@ func (h *Helper) updateConfig(configDir string, opt *StartOptions) error {
 		cfg.AdmissionConfig.PluginConfig[defaultsapi.BuildDefaultsPlugin] = buildDefaultsConfig
 	}
 
+	if opt.ServiceCatalog {
+		cfg.TemplateServiceBrokerConfig = &configapi.TemplateServiceBrokerConfig{
+			TemplateNamespaces: []string{OpenshiftNamespace},
+		}
+		if cfg.AssetConfig == nil {
+			cfg.AssetConfig = &configapi.AssetConfig{}
+		}
+		cfg.AssetConfig.ExtensionScripts = append(cfg.AssetConfig.ExtensionScripts, serviceCatalogExtensionPath)
+
+		extension := `
+window.OPENSHIFT_CONSTANTS.ENABLE_TECH_PREVIEW_FEATURE = {
+  service_catalog_landing_page: true,
+  template_service_broker: true,
+  pod_presets: true
+};
+`
+		extensionPath := filepath.Join(configDir, "master", "servicecatalog-extension.js")
+		err = ioutil.WriteFile(extensionPath, []byte(extension), 0644)
+		if err != nil {
+			return err
+		}
+		err = h.hostHelper.UploadFileToContainer(extensionPath, serviceCatalogExtensionPath)
+		if err != nil {
+			return err
+		}
+
+		// setup the api aggegrator needed by the service catalog
+		cfg.AggregatorConfig = configapi.AggregatorConfig{
+			ProxyClientInfo: configapi.CertInfo{
+				CertFile: aggregatorCert,
+				KeyFile:  aggregatorKey,
+			},
+		}
+		cfg.AuthConfig.RequestHeader = &configapi.RequestHeaderAuthenticationOptions{
+			ClientCA:            aggregatorCACert,
+			ClientCommonNames:   []string{"aggregator-front-proxy"},
+			UsernameHeaders:     []string{"X-Remote-User"},
+			GroupHeaders:        []string{"X-Remote-Group"},
+			ExtraHeaderPrefixes: []string{"X-Remote-Extra-"},
+		}
+
+		cacertPath := filepath.Join(configDir, aggregatorCACert)
+		cakeyPath := filepath.Join(configDir, aggregatorCAKey)
+		caserialPath := filepath.Join(configDir, aggregatorCASerial)
+		certPath := filepath.Join(configDir, aggregatorCert)
+		keyPath := filepath.Join(configDir, aggregatorKey)
+
+		out, err := localcmd.New("oadm").Args(
```

Comment:
i took the shortcut of using "oc adm" for now, so still not ideal but should be sufficient for what we need here.

the horrid thing sounds horrid and i don't think it's crazy for "oc cluster up" to require you have an "oc" command on your path :)